yarn.lock

```diff
@@ -12692,259 +12737,1482 @@ __metadata:
+  languageName: node
+  linkType: hard
+
+"@rushstack/ts-command-line@npm:4.10.10":
+  version: 4.10.10
+  resolution: "@rushstack/ts-command-line@npm:4.10.10"
+  dependencies:
+    "@types/argparse": 1.0.38
+    argparse: ~1.0.9
+    colors: ~1.2.1
+    string-argv: ~0.3.1
+  checksum: e2d47cbe6df4c47e297ef84b83f1f6f2306ae14cd765066ea966f3e6ed48e1382d20c942a324ff9e389f5e673a9e1ba477f5b9514303709fcf85f28e14cdd26c
+  languageName: node
+  linkType: hard
+
+"@segment/analytics-core@npm:1.0.1":
+  version: 1.0.1
+  resolution: "@segment/analytics-core@npm:1.0.1"
+  dependencies:
+    tslib: ^2.4.0
+  checksum: 7067c75fd4f567a2bf44718a307ca5a292b95d26c64c7c04e4f6202938cdddd7da0077d855ceba93ee7b820ce897e40fb80d8038c14ffbcc071cf5bf516d2847
+  languageName: node
+  linkType: hard
+
+"@segment/analytics-next@npm:^1.43.0":
+  version: 1.43.0
+  resolution: "@segment/analytics-next@npm:1.43.0"
+  dependencies:
+    "@lukeed/uuid": ^2.0.0
+    "@segment/analytics-core": 1.0.1
+    "@segment/analytics.js-video-plugins": ^0.2.1
+    "@segment/facade": ^3.4.9
+    "@segment/tsub": ^0.1.12
+    dset: ^3.1.2
+    js-cookie: 3.0.1
+    node-fetch: ^2.6.7
+    spark-md5: ^3.0.1
+    tslib: ^2.4.0
+    unfetch: ^4.1.0
+  checksum: 1d63cfe5a121d81485170b091bcf5ec139c39a668036ee792fb9e4ea5905a8db7a7841a3068926d2d7b91d035db9500676805b318275347fc2563a566a4c2c37
+  languageName: node
+  linkType: hard
+
+"@segment/analytics.js-video-plugins@npm:^0.2.1":
+  version: 0.2.1
+  resolution: "@segment/analytics.js-video-plugins@npm:0.2.1"
+  dependencies:
+    unfetch: ^3.1.1
+  checksum: 284b42bb05569366ac4a7c51c90d64bcea64315b2dd330ff8b58c63fa11949443b3d2f7e92d2e3c118495e9018352249183f7d3fb05e4b904148302536d385ba
+  languageName: node
+  linkType: hard
+
+"@segment/facade@npm:^3.4.9":
+  version: 3.4.9
+  resolution: "@segment/facade@npm:3.4.9"
+  dependencies:
+    "@segment/isodate-traverse": ^1.1.1
+    inherits: ^2.0.4
+    new-date: ^1.0.3
+    obj-case: 0.2.1
+  checksum: c6e139fd16464a1adb204bc13b958c95c96e5fa82c7281a4e71be4589b1a50a1d072f41d2bd8533c5fe1891e5a124af8c856ce26c493a0c5f548dff80c5ce572
+  languageName: node
+  linkType: hard
+
+"@segment/isodate-traverse@npm:^1.1.1":
+  version: 1.1.1
+  resolution: "@segment/isodate-traverse@npm:1.1.1"
+  dependencies:
+    "@segment/isodate": ^1.0.3
+  checksum: 06f783623c74a7a2310bdcca50355a1f7b9cbb7a1d77f0c7408def29bd0a355dbc2ff31516d025b7cd6baebfdca7d771fb62899f446366f86cd6d34cae9a7000
+  languageName: node
+  linkType: hard
+
+"@segment/isodate@npm:1.0.3, @segment/isodate@npm:^1.0.3":
+  version: 1.0.3
+  resolution: "@segment/isodate@npm:1.0.3"
+  checksum: 31faf12e83fa6ab5ba7dfd5600dfaf8d224a00ab24f3a2d3d9ba7f4ecdc1683a55c00fc9246585361aa5cb67c6d7846e1d5497cc3bcca58c105902b79c91c34a
+  languageName: node
+  linkType: hard
+
+"@segment/tsub@npm:^0.1.12":
+  version: 0.1.12
+  resolution: "@segment/tsub@npm:0.1.12"
+  dependencies:
+    "@stdlib/math-base-special-ldexp": ^0.0.5
```

Comment:
This adds a huge amount of dependencies, do you think it's possible to replace this in the `@segment` end? Both for the usage in this repo but also for all other users of the package. If not I'm wondering if we could perhaps host this package in a separate repo? Not too happy about bringing in this many new dependencies.